Typescript fails in `extend-expect.d.ts`

## Describe the bug
We use Typescript in our project. We do not have `skipLibCheck` enabled. 
After installation of RNTL v 12.4.0 and following [RNTL Docs on installation](https://callstack.github.io/react-native-testing-library/docs/getting-started#installation) it turns out that Typescript fails in RNTL's `node_modules/@testing-library/react-native/src/matchers/extend-expect.d.ts` file. 


<!--
  Describe your issue in detail. Include screenshots if needed. If this is a regression, let us know.
-->





## Expected behavior
Typescript does not fail

<!--
  A clear and concise description of what you expected to happen.
-->

## Steps to Reproduce

Here's a reproduction repo: TBD

<!--
  Let us know how to reproduce the issue. Include a code sample or share a project that reproduces the issue.
  Please follow the guidelines for providing a minimal example: https://stackoverflow.com/help/mcve.
-->

## Screenshots


<img width="1245" alt="image" src="https://github.com/callstack/react-native-testing-library/assets/36896640/6a24c75a-c4e2-4b11-80b6-3b768769552c">




<!--
  If applicable, add screenshots to help explain your problem.
-->

## Versions

<!--
  run following command in terminal of your root project and paste the result down

  `npx envinfo --npmPackages react,react-native,react-test-renderer,@testing-library/react-native`
-->


```bash
npmPackages:
    @testing-library/react-native: ^12.4.0 => 12.4.0
    react: 18.2.0 => 18.2.0
    react-native: ^0.72.7 => 0.72.7
    react-test-renderer: 18.2.0 => 18.2.0
```

Yeah, clone the repro repo, check out [this commit](https://github.com/erheron/rntl-1534-repro/commit/fa81d120cd3354526ceeb272d2f0f695dfcfccf1) and run `yarn tsc`. What you should see is exactly this:

```bash
node_modules/@testing-library/react-native/src/matchers/extend-expect.d.ts:3:50 - error TS2307: Cannot find module '../matches' or its corresponding type declarations.

3 import type { TextMatch, TextMatchOptions } from '../matches';
                                                   ~~~~~~~~~~~~

node_modules/@testing-library/react-native/src/matchers/extend-expect.d.ts:4:48 - error TS2307: Cannot find module '../helpers/matchers/match-accessibility-value' or its corresponding type declarations.

4 import type { AccessibilityValueMatcher } from '../helpers/matchers/match-accessibility-value';
                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@testing-library/react-native/src/matchers/extend-expect.d.ts:5:28 - error TS2307: Cannot find module './to-have-style' or its corresponding type declarations.

5 import type { Style } from './to-have-style';
                             ~~~~~~~~~~~~~~~~~


Found 3 errors in the same file, starting at: node_modules/@testing-library/react-native/src/matchers/extend-expect.d.ts:3

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

--- a/src/matchers/extend-expect.d.ts
+++ b/src/matchers/extend-expect.d.ts
@@ -1,8 +1,5 @@
 import type { StyleProp } from 'react-native';
 import type { ReactTestInstance } from 'react-test-renderer';
-import type { TextMatch, TextMatchOptions } from '../matches';
-import type { AccessibilityValueMatcher } from '../helpers/matchers/match-accessibility-value';
-import type { Style } from './to-have-style';
 
 export interface JestNativeMatchers<R> {
   toBeOnTheScreen(): R;
@@ -38,3 +35,22 @@ declare module '@jest/expect' {
   interface Matchers<R extends void | Promise<void>>
     extends JestNativeMatchers<R> {}
 }
+
+// Used types
+
+export type Style = ViewStyle | TextStyle | ImageStyle;
+
+export interface AccessibilityValueMatcher {
+  min?: number;
+  max?: number;
+  now?: number;
+  text?: TextMatch;
+}
+
+export type TextMatch = string | RegExp;
+export type TextMatchOptions = {
+  exact?: boolean;
+  normalizer?: NormalizerFn;
+};
+
+export type NormalizerFn = (textToNormalize: string) => string;

